### ISSUE TITLE
Active projects filter

### ISSUE THREAD

**Issue by Robadob** [RSE-Sheffield/RSE-Sheffield.github.io#454]
It would be good to have a sub-page of [projects](https://rse.shef.ac.uk/service/projects/), e.g. `https://rse.shef.ac.uk/service/projects/active` which had the list with completed projects hidden.

Not sure how feasible this would be via Jekyll, but it could be useful in cases where we want to talk about what we're actively doing. e.g. @bobturneruk's application for a summer intern, who might help out on one or more of our active projects.

**Comment by ptheywood** [comment]
As the data includes start and end date it would be relatively simple to at build-time compare to the current start and end date. 

Alternatively outputting those as `data` attributes on the containing html element and a bit of js to hide ones outside the range would be viable (i.e. how the upcoming events filter works)

**Comment by Robadob** [comment]
Yeah I suppose with JS, it would be possible to use `#active` in the url to hyperlink to active projects, or even filter via the tag cloud listed at the top of the page. More work though.

**Comment by Robadob** [comment]
https://rse-mirror.robadob.org/service/projects/active

Very quick clone of the existing project page, to list only active ones. Would need some revision and integration with the other pages though. Although, as a temporary fix adding hidden active/completed projects pages could suffice.

Not too sure how we'd make some kind of auto filter via the `/active` part to filter via tags. If we can somehow forward invalid 'slugs'(?) to a variable, it might not be too difficult. Although as Jekyll is a static site, I guess that's not possible without creating a dedicated `.md` for each tag. Which leaves us with the JS filtering.

**Comment by Robadob** [comment]
Ok I've made some progress.

I've updated the project divs so they now look like this

`<div class="project completed tag-Python  tag-Scrum  tag-testing  tag-machine-learning  tag-Gaussian-processes  tag-containers  tag-cloud  tag-continuous-integration">`


The tag cloud has also been updated to links of this format.

`<a class="tag-link" href="#C">C</a>`

So this jquery now sets them up to hide all other tags.



```js
$("a.tag-link").on( "click", function(e) {
    e.preventDefault();
    var tag = $(this).attr("href").substring($(this).attr("href").lastIndexOf('#') + 1);
    $("a.tag-link.selected").removeClass("selected");
    $(this).addClass("selected");
    $('.project.tag-' + tag).show();
    $('.project:not(.tag-' + tag + ')').hide();
});
```

https://rse-mirror.robadob.org/service/projects/

**Note:** Currently escaping tag spaces to `-`, should probably escape `#` as i'm not sure it wouldn't break if we had a C# project.
